OsmAnd/src/net/osmand/plus/quickaction/actions/AddPOIAction.java

```diff
@@ -79,6 +80,29 @@ public AddPOIAction(QuickAction quickAction) {
 		super(quickAction);
 	}
 
+	@Override
+	public int getIconRes(Context context) {
+		if (context instanceof MapActivity) {
+			final OsmandApplication application = (OsmandApplication) (context).getApplicationContext();
+			final MapPoiTypes poiTypes = application.getPoiTypes();
+			final Map<String, PoiType> allTranslatedNames = poiTypes.getAllTranslatedNames(true);
```

Comment:
I suggest to extract poiTypes  and allTranslatedNames to class field. Do the same in drawUI method.

Reply:
Where is the best place to initialize them?